### PR TITLE
Fix bug and add initializing fallback

### PR DESCRIPTION
**Pull request by ckybonist**
## Background & Motivation
Recently, our company receives thousands of glightbox errors from old browser, for instance, iOS safari < 10.

Related issue: https://github.com/biati-digital/glightbox/issues/164

## Bug Description
In [each() helper method](https://github.com/biati-digital/glightbox/blob/master/src/js/utils/helpers.js#L40-L68), if the `collection` is `NodeList`, It will be treated as `Object` so that the the method using `for-in` to loop over the NodeList which introduced 2 types of errors, e.g. 

Initialization error:
1. `Cannot assign to read only property '1604491181466' of 8` ([1604491181466 is the `uid`](https://github.com/polydice/glightbox/blob/59e5eafe6c6debdba505f844e7856ef88f80be63/src/js/utils/helpers.js#L1))
2. `Attempted to assign to readonly property.`

onClick error (which means the error will be thrown every time when user clicking the image even if the glightbox was failed to init)
1. `undefined is not an object (evaluating 'this.elements.length')`

## Changes
- Add a function to check the object is `NodeList`
- Using IE compatible way to iterate the NodeList
- Fallback mechanism: 
    - Try-catch the instance initialization error
    - Set the `glightbox` variable to `undefined` on error so that user could further implement other fallbacks of their own logic which related to glightbox 

## Test Result
- Works fine in iOS 9

image

- Got a init errors in iOS 8, however clicking the image will fallback to `href` behavior rather than getting errors

## References
https://developer.mozilla.org/en-US/docs/Web/API/NodeList#Example

### CODE REVIEW

File: src/js/utils/helpers.js [hunk 1]
```diff
@@ -48,10 +48,16 @@ export function each(collection, callback) {
         return;
     }
 
-    if (isArrayLike(collection) && !isObject(collection)) {
-        let l = collection.length,
-            i = 0;
-        for (; i < l; i++) {
+    if (isNodeList(collection)) {
```
Comment (biati-digital) [inline src/js/utils/helpers.js:51]: the current implementation uses a for loop to be able to use break, with your code that will not be possible.

**Comment by biati-digital** [comment]
Hi, thanks for the pull request, i've added a comment, also please follow the same coding style, no space between function identifiers and 4 spaces for indentation

File: dist/js/glightbox.js [hunk 2]
```diff
@@ -119,11 +119,12 @@
...
 
-    if (isArrayLike(collection) && !isObject(collection)) {
-      var l = collection.length,
-          i = 0;
-
-      for (; i < l; i++) {
+    if (isNodeList(collection)) {
+      Array.prototype.forEach.call(collection, function (item, i) {
+        callback.call(item, item, i, collection);
+      });
```
Comment (ckybonist) [inline dist/js/glightbox.js:125]: Did you mean this block could not use `break`?

File: dist/js/glightbox.js [hunk 3]
```diff
@@ -119,11 +119,12 @@
...
+    } else if (isArrayLike(collection) && !isObject(collection)) {
+      for (var i = 0; i < collection.length; i++) {
```
Comment (ckybonist) [inline dist/js/glightbox.js:127]: Or I should revert this line back?

**Review by ckybonist — COMMENTED** [review]
Leave few questions about https://github.com/biati-digital/glightbox/pull/165#discussion_r518448313
@biati-digital

File: src/js/utils/helpers.js (hunk 1 above)
Comment (ckybonist) [reply src/js/utils/helpers.js:51]: Revise coding style at https://github.com/biati-digital/glightbox/pull/165/commits/a8524327f6628d2451d9aeac9b844bce59b41993

**Comment by ckybonist** [comment]
Any update here?

**Pull request closed without merging**